File: run_image.py
"""Run one raw MIN0 CORE FORTH v0.1 image and print a machine-readable result."""

from __future__ import annotations

import argparse
import json
from pathlib import Path

from min0_core_forth_vm import Min0CoreForthVM


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("image", type=Path)
    args = parser.parse_args()
    vm = Min0CoreForthVM()
    vm.load(args.image.read_bytes())
    stack = vm.run()
    print(json.dumps({"stack": stack, "steps": vm.steps}, separators=(",", ":")))


if __name__ == "__main__":
    main()

File: run_image.rb
# frozen_string_literal: true

require "json"
require_relative "min0_core_forth_vm"

abort "usage: ruby run_image.rb IMAGE" unless ARGV.length == 1

vm = Min0CoreForth::VM.new
vm.load(File.binread(ARGV.fetch(0)))
stack = vm.run
puts JSON.generate({ stack: stack, steps: vm.steps })
